include/asio/ssl/dtls/acceptor.hpp

```diff
@@ -767,6 +768,7 @@ class acceptor
       , ah_(std::move(ah))
       , sock_(sock)
       , buffer_(buffer)
+      , work_(associated_executor<AcceptHandler>::get(ah))
```

Comment:
hmm...no, I don't think this is right. The work should be using the executor of the I/O object and not the executor associated with the completion handler. The completion handler's executor will automatically get a work guard when you submit your own composed operation to an Asio I/O operation (I think you are calling the datagram socket's `async_receive_from`?)

Asio (and therefore Networking TS) require  **two** executor work guards. One of them you get for "free" when you call an I/O operation, and the other one you have to provide yourself.